src/main/scala/tech/cryptonomic/conseil/routes/Data.scala

```diff
@@ -0,0 +1,48 @@
+package tech.cryptonomic.conseil.routes
+
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import de.heikoseeberger.akkahttpjackson.JacksonSupport
+import tech.cryptonomic.conseil.generic.chain.DataPlatform
+import tech.cryptonomic.conseil.generic.chain.DataTypes.Query
+import tech.cryptonomic.conseil.util.RouteHandling
+
+import scala.concurrent.ExecutionContext
+
+/** Companion object providing apply implementation */
+object Data {
+  def apply(config: Config)(implicit ec: ExecutionContext): Data = new Data(config, DataPlatform())
+}
+
+/**
+  * Platform discovery routes.
+  *
+  * @param queryProtocolPlatform QueryProtocolPlatform object which checks if platform exists and executes query
+  * @param apiExecutionContext   is used to call the async operations exposed by the api service
+  */
+class Data(config: Config, queryProtocolPlatform: DataPlatform)(implicit apiExecutionContext: ExecutionContext)
+  extends LazyLogging with RouteHandling with JacksonSupport {
+
+  val route: Route =
+    get {
+      pathPrefix(Segment) { platform =>
+        pathPrefix(Segment) { network =>
```

Comment:
I don't exactly remember if the routing api allows to pattern match on multiple segments to extract more than one parameter at once, simplifying such nested cases

Reply:
I don't think it's necessary in this case - it is simple and readable ;)